app/services/ambassador_task_assignment_creator.rb

```diff
@@ -0,0 +1,43 @@
+module AmbassadorTaskAssignmentCreator
```

Comment:
Why create this module instead of putting these directly into the worker?

Reply:
It simplifies testing and REPL-driven development somewhat to be able to execute the service methods separately, but mainly because SRP. Keeping logic out of workers (especially when the workers are life-cycle based as these are) adds flexibility. For example, if you want to combine different sets of business logic in the context of a job, that's a simpler addition when the guts of the logic are elsewhere.